### ISSUE TITLE
glob dependency issue

### ISSUE THREAD

**Issue by jonathanKingston** [ember-cli/ember-cli#3492]
Since today rebuilding apps has been getting this:
```
Livereload server on port 35729
Serving on http://0.0.0.0:4200/
undefined is not a function
TypeError: undefined is not a function
    at rimraf (/home/jonathan/emberApp/node_modules/ember-cli/node_modules/broccoli-caching-writer/node_modules/rimraf/rimraf.js:57:13)
    at lib$rsvp$node$$tryApply (/home/jonathan/emberApp/node_modules/ember-cli/node_modules/broccoli-caching-writer/node_modules/rsvp/dist/rsvp.js:1467:11)
    at lib$rsvp$node$$handleValueInput (/home/jonathan/emberApp/node_modules/ember-cli/node_modules/broccoli-caching-writer/node_modules/rsvp/dist/rsvp.js:1567:20)
    at fn (/home/jonathan/emberApp/node_modules/ember-cli/node_modules/broccoli-caching-writer/node_modules/rsvp/dist/rsvp.js:1555:18)
    at /home/jonathan/emberApp/node_modules/ember-cli/node_modules/broccoli-caching-writer/index.js:100:14
    at lib$rsvp$$internal$$tryCatch (/home/jonathan/emberApp/node_modules/ember-cli/node_modules/promise-map-series/node_modules/rsvp/dist/rsvp.js:489:16) 
    at lib$rsvp$$internal$$invokeCallback (/home/jonathan/emberApp/node_modules/ember-cli/node_modules/promise-map-series/node_modules/rsvp/dist/rsvp.js:501:17)
    at lib$rsvp$$internal$$publish (/home/jonathan/emberApp/node_modules/ember-cli/node_modules/promise-map-series/node_modules/rsvp/dist/rsvp.js:472:11)  
    at lib$rsvp$asap$$flush (/home/jonathan/emberApp/node_modules/ember-cli/node_modules/promise-map-series/node_modules/rsvp/dist/rsvp.js:1290:9)
    at process._tickCallback (node.js:355:11)
```

The fix is either drop glob out of the bundledDependencies or upgrade glob to 4.2.1+

### Long term fix
I'm starting also to think that bundledDependencies shouldn't be used in conjunction with non explicit package.json dependencies, this has come about because broccoli-caching-writer has updated rimraf to point at a version that is newer than the native API. Bundling also seems to look only for the first nearest matching dependency (I'm not sure if this is intentional).
Further debugging required.

**Comment by rwjblue** [comment]
See:

https://github.com/ember-cli/ember-cli/issues/3486#issuecomment-78039493
https://github.com/ember-cli/ember-cli/issues/3445
https://github.com/ember-cli/ember-cli/issues/3434
https://github.com/ember-cli/ember-cli/issues/3413#issuecomment-77480891
https://github.com/ember-cli/ember-cli/issues/3413
https://github.com/ember-cli/ember-cli/pull/3428

**Comment by rwjblue** [comment]
tldr; you have the following options:

* Update to 0.2.0
* Lock `rimraf@2.2.8` and `glob@4.0.5`

**Comment by rwjblue** [comment]
The longer term solution is to audit and fix all Broccoli plugins that use `glob` to ensure that the new `follow: true` option is passed.  Once that is done, we can unlock the deps here.

**Comment by jonathanKingston** [comment]
@rwjblue sorry missed those, I checked only open issues (hangs head in shame).

Is it worth opening a new issue to review and fix all of the plugins you mention?

**Comment by jonathanKingston** [comment]
Also seems a strong enough issue to add to common issues if that is ok?

**Comment by stefanpenner** [comment]
> Also seems a strong enough issue to add to common issues if that is ok?

yes, although unfortunate...

**Comment by rwjblue** [comment]
I want to keep this open to track the main issue of getting these deps unlocked.

**Comment by jonathanKingston** [comment]
:+1: @rwjblue.

**Comment by jdhines** [comment]
@rwjblue I am already on ember-cli 0.2.0-beta.1 and still have this problem with a new Ember app.

**Comment by stefanpenner** [comment]
> @rwjblue I am already on ember-cli 0.2.0-beta.1

we are at `0.2.3` can you upgrade?

**Comment by jdhines** [comment]
Movin' faster than I can keep up. I can in my development; harder in my real app since I have to go through and diff so many files during init.
